File: src/rearrange/rearrange_2_gs.cc
#include <iostream>

int w( int N ) {
	return __builtin_popcount(N);
}

bool check( int N, int T ) {
	return (N >= 2 * ((8*T + 2 - 2*w(T)) / 3));
}

int main( ) {
	int T = 1;
	for( int N = 1; true; N++ ) {
		while( check(N,T) ) {
			T++;
		}
		std::cout << 2 << " " << N << " " << T << "\n";
	}

	return 0;
}

File: src/rearrange/rearrange_2_gs.py
def w(N):
    return bin(N).count("1") # Or in Python 3.8+: N.bit_count()

def check(N,T):
    return N >= 2 * ((8*T + 2 - 2*w(T)) // 3)

N = 1
T = 1
while True:
    while check(N,T):
        T += 1
    print(2,N,T)
    N += 1
